### ISSUE TITLE
CVC4 not respecting --tlimit

### ISSUE THREAD

**Issue by HanielB** [CVC4/cvc4-projects#62]
Sometimes CVC4 does not respect, by a significant margin, the time limit given by `--tlimit`. A complete list of SMT-LIB benchmarks (as of Mar 2019) on which CVC4, in default mode, surpasses the given time, 5s, but does not terminate on its own, is below. The file names determine for how long CVC4 was allowed to run (e.g., bad-7s.csv has the benchmarks on which CVC4 did not terminate on its own before 7s). Experiments ran in the Barrett's cluster.

[bad-tlimit.zip](https://github.com/CVC4/CVC4/files/3024063/bad-tlimit.zip)

Related issues are CVC4/CVC4#2902 and CVC4/CVC4#2438

**Comment by ajreynol** [comment]
Two more related issues that seems to have lost connection after transferring to cvc4-projects:
https://github.com/CVC4/CVC4/issues/3040
https://github.com/CVC4/CVC4/issues/1010

**Comment by nafur** [comment]
These issues should be fixed with https://github.com/CVC4/CVC4/commit/a0a389a1f7c16b6a1fb56b3ce8ee519cf5717f04. Do you have an easy way of redoing these experiments?

**Comment by HanielB** [comment]
It was actually a simple experiment. I ran on the whole non-incremental SMT-LIB with `--tlimit 5000`, 20s timeout, 8gb RAM, and generated the CSVs with

```compare-smt-runs.py --time --csv default/ > bad-20s.csv
compare-smt-runs.py --time --csv --to 15 default/ > bad-15s.csv
compare-smt-runs.py --time --csv --to 10 default/ > bad-10s.csv
compare-smt-runs.py --time --csv --to 7 default/ > bad-7s.csv
compare-smt-runs.py --time --csv --to 5 default/ > bad-5s.csv
```

**Comment by nafur** [comment]
Fixed with CVC4/CVC4#4655, I repeated these experiments and didn't see a single run that took more than six seconds.
